interface/src/app/volcano-plot/volcano-plot.js

```diff
@@ -1,35 +1,23 @@
 /*
  * The volcano-plot module provides a directive and controller for embedding
- * volcano plots in other views. Requires samples to be grouped in the
- * adage.analyze.sampleBin in order to function properly -- if no samples are
- * chosen or placed into groups, an error will be displayed.
+ * volcano plots in other views. Requires properly prepared data to be specified
+ * via plotData.
  */
 angular.module('adage.volcano-plot', [
   'ngVega',
   'greenelab.stats',
-  'adage.analyze.sampleBin',
   'adage.volcano-plot-vgspec'
 ])
 
-.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec', 'SampleBin',
-  function VolcanoPlotCtrl($scope, VolcanoPlotSpec, SampleBin) {
-    // retrieve the nodeInfo map (for node names) via the SampleBin service
-    // $scope.nodeInfo = SampleBin.getNodeInfo(); // TBD
-    $scope.nodeInfo = [
-      {'id': 1, 'name': 'Node1pos'},
-      {'id': 2, 'name': 'Node2pos'},
-      {'id': 518, 'name': 'test name'}
-    ];
-
+.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec',
+  function VolcanoPlotCtrl($scope, VolcanoPlotSpec) {
     // define controller instance variables that link with vega
-    this.data = {
-      source: $scope.plotData
-      // FIXME: this is not working as it should to bring in node names
-      // nodes: $scope.nodeInfo
-      // FIXME: can't get selectedNodes out this way... need to use view api?
-      // selectedNodes: $scope.selectedNodes
-    };
     this.spec = VolcanoPlotSpec;
+    this.data = $scope.plotData;
```

Comment:
Thank you @mhuyck for the explanation. I think another option is to use a link function inside "volcanoPlot" directive instead of the "VolcanoPlotCtrl" controller to introduce "VolcanoPlotSpec" service inside the scope of "volcanoPlot", like this:
```
.directive('volcanoPlot', ['VolcanoPlotSpec', function(VolcanoPlotSpec) {
  return {
    restrict: 'E',
    scope: {
      plotData: '=',
      selectedNodes: '='
    },
    templateUrl: 'volcano-plot/volcano-plot.tpl.html',
    link: function($scope) {
      $scope.spec = VolcanoPlotSpec;
    }
  };
}]);
```
then you can refer to the spec and data directly in **volcano-plot.tpl.html** like this:
```
<div vega spec="spec" vega-data="plotData"></div>
```
I agree that using `component` will be a much better solution.